GraphResponse.md

```diff
@@ -7,6 +7,206 @@ Provide a decorator for the native response message to enable attaching Graph sp
+ 2. a ServiceException in the case the call fails.
+
+We consider this a design flaw as the client library swallows useful information such as the response headers and the http status code. Mitigation such as stuffing the information in AdditionalData is insufficient for all scenarios. 
+For example, there is no AdditionalData on Stream objects, and Delete operations return void.
+
+Related to:
+
+- https://github.com/microsoftgraph/msgraph-sdk-dotnet/issues/457
+- https://github.com/microsoftgraph/msgraph-sdk-dotnet/issues/361
+
+
+## Current Implementations
+
+### Javascript
+
+The Javascript SDK exposes the native [Response](https://developer.mozilla.org/en-US/docs/Web/API/Response) object when selecting the RAW ResponseType option.
+
+### .Net
+
+> Currently not supported
+
+### Java
+
+> Currently not supported. Should be able to use the same pattern as .Net.
+
+### PHP
+
+The PHP SDK returns a Graph model (or collection) or a GraphResponse object, depending on whether a returnType is specified. This means that the returnType information is not supplied to the GraphResponse. The response body is set as a JSON array on GraphResponse. The only work I'd suggest here is that we add an option to pass the returnType to the GraphResponse and add a lazy deserialize method so it is easy to get the Graph models.
+
+## The GraphResponse Object
+
+The GraphResponse object can wrap the native HttpResponseMessage object so that customers can use the familiar HttpResponseMessage members. 
+GraphResponse will be part of Microsoft.Graph.Core and it will expose the headers,Content and status code while also keeping reference to the Request object as below.
+
+```csharp
+
+    /// <summary>
+    /// The GraphResponse Object
+    /// </summary>
+    public class GraphResponse 
+    {
+        /// <summary>
+        /// The GraphResponse Constructor
+        /// </summary>
+        /// <param name="iBaseRequest">The Request made for the response</param>
+        /// <param name="httpResponseMessage">The response</param>
+        public GraphResponse(IBaseRequest iBaseRequest,HttpResponseMessage httpResponseMessage)
+        {
+            this.httpResponseMessage = httpResponseMessage;
+            this.BaseRequest = iBaseRequest;
+        }
+
+        private readonly HttpResponseMessage httpResponseMessage;
+        
+        /// <summary>
+        /// The Response Status code
+        /// </summary>
+        public HttpStatusCode StatusCode => httpResponseMessage.StatusCode;
+        
+        /// <summary>
+        /// The Response Content
+        /// </summary>
+        public HttpContent Content => httpResponseMessage.Content;
+        
+        /// <summary>
+        /// The Response Headers
+        /// </summary>
+        public HttpResponseHeaders HttpHeaders => httpResponseMessage.Headers;
+
+        /// <summary>
+        /// The reference to the Request
+        /// </summary>
+        public IBaseRequest BaseRequest;
+
```

Comment:
```csharp
// Deserialize the response
public T GetResponseObject<T>()
```

ResponseHandler is on BaseRequest but not IBaseRequest, so we'd want to change BaseRequest property to access BaseRequest reference from the request.

Reply:
Since this work is planned to be done on a breaking change, we can finally make ResponseHandler available on IBaseRequest with the IResponseHandler interface. Yes?

Reply:
Yes, we can do that as long as we implement the matching change in the generator. Please add a work item to track that if we don't do this immediately. Otherwise, please implement the change in a matching feature/2.0 branch in the generator.

Reply:
Okey dokey.

Line 167 covers the change that will be made to BaseRequest as you described.